lib/galaxy/jobs/actions/post.py

```diff
@@ -146,8 +146,10 @@ def execute(cls, app, sa_session, action, job, replacement_dict):
                 for input_assoc in job.input_dataset_collections:
                     if input_assoc.name == input_file_var:
                         if input_assoc.dataset_collection:
-                            hdca = input_assoc.dataset_collection
-                            replacement = hdca.name
+                            # Either a HDCA or a DCE - only HDCA has a name.
+                            has_collection = input_assoc.dataset_collection
+                            if hasattr(has_collection, "name"):
+                                replacement = has_collection.name
```

Comment:
I'd write it as:
```python
                    if input_assoc.name == input_file_var:
                        # Either a HDCA or a DCE - only HDCA has a name.
                        has_collection = input_assoc.dataset_collection
                        if has_collection and hasattr(has_collection, "name"):
                            replacement = has_collection.name
```

Reply:
Yes - a lot better - I missed the context here thanks!